### DISCUSSION TITLE
Broadcast or share reactive state between pages in multipage app

### DISCUSSION THREAD

**Discussion by HugoP** [widgetti/solara#659] [Q&A]
Hi,

I have used two of the examples to help with creating a multipage live update app.

[Live Update](https://solara.dev/documentation/examples/general/live_update) and [Multiple scripts](https://solara.dev/documentation/advanced/howto/multipage)

The live part of the app is working by subscribing to a MQTT topic and parsing the payload before rendering the live data.

Is there a way to broadcast or inform another page in the app when a new topic has arrived and render the new value on the that page?

At the end of the day the required data ends up in a PostgreSQL database and the MQTT subscribe topic acts like a heart beat to inform the app that new data is available.

Kind regards

**Comment by HugoP** [comment]
Looks like this might be the trick.

Examples
[Building Your First Python Package: A Step-by-Step Guide](https://clouddevs.com/python/building-your-first-package/)
[How to Build a Complete Python Package Step-by-Step](https://www.youtube.com/watch?v=5KEObONUkik)

It's tough being a noob!

**Comment by maartenbreddels** [comment]
Hi Hugo,

I guess you moved on, but this example
 * https://solara.dev/apps/multipage

Show an example of a reactive variable that is used my multiple pages, see source:
 * https://github.com/widgetti/solara/blob/master/solara/website/pages/apps/multipage/__init__.py

I hope that still helps or future users.

Regards,

Maarten

**Comment by HugoP** [comment]
Hi @maartenbreddels ,

Thanks for the reply.

I seem to have missed that example and followed this approach

```bash
solara create button multipage-demo/01-click-button.py
solara create markdown multipage-demo/02-markdown-editor.py
solara run ./multipage-demo
```

Your example looks like a simpler way of doing what I need.

Regards
Hugo

**Reply by HugoP** [reply]
I made a copy of just the multipage folder from  the github repo to start using it as a example and running it like so:

```bash
solara run ./multipage
```
But I'm getting errors

```bash
from . import SharedComponent
ImportError: attempted relative import with no known parent package
(.venv) root@host:~/code/solara_play$ solara run ./multipage --host=172.26.1.205
```

PS. I would really like to use the __init__.py file technique to turn a folder into a package and it works on other tests I have done when solara is not involved.

What am I doing wrong?

Regards
Hugo

**Reply by iisakkirotko** [reply]
Hey @HugoP! If you already have your app installed as a package, you can run it using `solara run multipage`. Note that you shouldn't make the argument a path with `./`. Sometimes it also helps to have your app in a subdirectory (`./multipage/app`) and then run it with `solara run multipage.app`, but that depends on the rest of your package structure.
